pkg/controller/labelidentity/label_group_index.go

```diff
@@ -0,0 +1,503 @@
+	return s.selector.NormalizedName
+}
+
+// labelIdentityMatch is constructed from a LabelIdentity and used for matching
+// between LabelIdentity and selectorItems. It also stores the current selectorItems
+// that matches this LabelIdentity.
+type labelIdentityMatch struct {
+	id               uint32
+	namespace        string
+	namespaceLabels  map[string]string
+	podLabels        map[string]string
+	selectorItemKeys sets.String
+}
+
+// matches knows if a LabelIdentity matches a selectorItem.
+func (l *labelIdentityMatch) matches(s *selectorItem) bool {
+	selectorItemNamespace := s.selector.Namespace
+	if selectorItemNamespace != emptyNamespace && selectorItemNamespace != l.namespace {
+		return false
+	}
+	if s.selector.NamespaceSelector != nil && !s.selector.NamespaceSelector.Matches(labels.Set(l.namespaceLabels)) {
+		return false
+	}
+	// At this stage Namespace has matched
+	if s.selector.PodSelector != nil {
+		return s.selector.PodSelector.Matches(labels.Set(l.podLabels))
+	}
+	// SelectorItem selects all when all selectors are missing.
+	return true
+}
+
+// constructMapFromLabelString parses label string of format "app=client,env=dev" into a map.
+func constructMapFromLabelString(s string) map[string]string {
+	m := map[string]string{}
+	kvs := strings.Split(s, ",")
+	for _, kv := range kvs {
+		kvpair := strings.Split(kv, "=")
+		m[kvpair[0]] = kvpair[1]
+	}
+	return m
+}
+
+// newLabelIdentityMatch constructs a labelIdentityMatch from a normalized LabelIdentity string.
+func newLabelIdentityMatch(labelIdentity string, id uint32) *labelIdentityMatch {
+	nsIndex, podIndex := labelRegex.SubexpIndex("nslabels"), labelRegex.SubexpIndex("podlabels")
+
+	labelMatches := labelRegex.FindStringSubmatch(labelIdentity)
+	nsLabels := constructMapFromLabelString(labelMatches[nsIndex])
+	podLabels := constructMapFromLabelString(labelMatches[podIndex])
+
+	namespace := nsLabels[apiv1.LabelMetadataName]
+	return &labelIdentityMatch{
+		id:               id,
+		namespace:        namespace,
+		namespaceLabels:  nsLabels,
+		podLabels:        podLabels,
+		selectorItemKeys: sets.NewString(),
+	}
+}
+
+// selectorItemKeyFunc knows how to get the key of a selectorItem.
+func selectorItemKeyFunc(obj interface{}) (string, error) {
+	sItem, ok := obj.(*selectorItem)
+	if !ok {
+		return "", fmt.Errorf("object is not of type *selectorItem: %v", obj)
+	}
+	return sItem.getKey(), nil
+}
+
+func newSelectorItemStore() cache.Indexer {
+	indexers := cache.Indexers{
+		cache.NamespaceIndex: func(obj interface{}) ([]string, error) {
+			sItem, ok := obj.(*selectorItem)
+			if !ok {
+				return []string{}, nil
+			}
+			// sItem.Selector.Namespace == "" means it's a cluster scoped selector, we index it as it is.
+			return []string{sItem.selector.Namespace}, nil
+		},
+		policyIndex: func(obj interface{}) ([]string, error) {
+			sItem, ok := obj.(*selectorItem)
+			if !ok {
+				return []string{}, nil
+			}
+			return sItem.policyKeys.List(), nil
```

Comment:
Use UnsortedList to avoid unnecessary overhead spent on sorting

Reply:
Done